min-turnup/phase1/azure/configure-vm.sh

```diff
@@ -0,0 +1,69 @@
+#! /bin/bash
+
+set -x
+set -o errexit
+set -o pipefail
+set -o nounset
+
+mkdir -p /etc/systemd/system/docker.service.d/
+cat <<EOF > /etc/systemd/system/docker.service.d/clear_mount_propagtion_flags.conf
+[Service]
+MountFlags=shared
+EOF
+cat <<EOF > /etc/systemd/system/docker.service.d/overlay.conf
+[Service]
+ExecStart=
+ExecStart=/usr/bin/docker daemon -H fd:// --storage-driver=overlay
+EOF
+
+curl -sSL https://get.docker.com/ | sh
+
+apt-get update
+#apt-get dist-upgrade -y
+apt-get install -y jq
+
+systemctl start docker || true
+
+ROLE="node"
+if [[ $(hostname) = *master* ]]; then
+	ROLE="master"
+fi
+
+azure_file="/etc/kubernetes/azure.json"
+config_file="/etc/kubernetes/k8s_config.json"
+
+mkdir /etc/kubernetes
+# these get filled in from terraform
+echo -n "${azure_json}" | base64 -d > "$azure_file"
+echo -n "${k8s_config}" | base64 -d > "$config_file"
+echo -n "${kubelet_tar}" | base64 -d > "/etc/kubernetes/kubelet.tar"
+echo -n "${root_tar}" | base64 -d > "/etc/kubernetes/root.tar"
+echo -n "${apiserver_tar}" | base64 -d > "/etc/kubernetes/apiserver.tar"
+
+MASTER_IP="$(cat "$config_file" | jq -r '.phase1.azure.master_private_ip')"
+
+jq ". + {\"role\": \"$ROLE\", \"master_ip\": \"$MASTER_IP\"}" "$config_file" > /etc/kubernetes/k8s_config.new; cp /etc/kubernetes/k8s_config.new "$config_file"
+
+mkdir -p /srv/kubernetes
+for bundle in root kubelet apiserver; do
+  cat "/etc/kubernetes/$bundle.tar" | sudo tar xv -C /srv/kubernetes
+done;
+
+installer_container_spec="$(cat "$config_file" | jq -r '.phase2.installer_container_spec')"
+
+cat << EOF > /etc/kubernetes/install.sh
+systemctl stop docker
```

Comment:
RE: this and the question above... I had done some hacky things in `configure-vm.sh`. In Azure, the `configure-vm.sh` script didn't seem to be available in plaintext (there might be a base64 copy somewhere). So what I was doing was writing the `install.sh` script out to disk.

So the `configure-vm.sh` dropped the `install.sh` script for future runs, as I was iterating on the Ansible container and then invoked it at the end of `configure-vm.sh` for the initial deployment.

The thinking here was that someone would want to upgrade their cluster. They've updated the `k8s_config.json` to reference the newer Kubernetes version, now they need to re-run the playbook container. With my setup, they just call `install.sh` again. Kubelet gets torn down as part of rebooting Docker. Playbook runs, new kubelet boots up. So `install.sh` is what I was treating as needing to be idempotent.

(and yes, it should have a shebang, if we want to keep this model. I don't know if `configure-vm.sh` is meant to be re-run on GCE?)